### DISCUSSION TITLE
Opening a Pluto notebook from the command line

### DISCUSSION THREAD

**Discussion by mgkuhn** [fonsp/Pluto.jl#2725] [Questions about using Pluto] [answered]
I normally start Pluto via this little Julia script, works on both Linux and macOS:

```
$ cat `which pluto`
#!/usr/bin/env -S julia -e include(popfirst!(ARGS))
# Fire up Pluto.jl and optionally open a notebook
import Pluto
if length(ARGS) == 0
    Pluto.run()
elseif length(ARGS) == 1
    Pluto.run(notebook=ARGS[1])
else
    println("usage: pluto [notebook.jl]")
    exit(1)
end
```

How do you start Pluto?

**Comment by mgkuhn** [comment]
Note that the way in which the kernel and `env -S` interpret command-line arguments in the [hashbang](https://en.wikipedia.org/wiki/Shebang_(Unix)) (`#!`) line differs slightly between Linux (does no parsing, passes on a single argv[1]) and macOS (attempts very basic shell-like command-line parsing), hence the careful omission of unnecessary shell meta-characters (other than space) in that line, to achieve portability across both these platforms.

**Comment by Pangoraw** [comment, marked as answer]
Here is mine (it's a bash function):

```bash
pluto() {
  PLUTO_ARGS="dismiss_update_notification=true, require_secret_for_open_links=true, require_secret_for_access=true"
  case "$2" in
      yes) REVISE="using Revise;" ;;
      *)   REVISE="" ;;
  esac
  if [[ -d "$HOME/Projects/Pluto.jl" ]]; then
    PLUTO_PROJECT=$HOME/Projects/Pluto.jl
  else
    PLUTO_PROJECT=@pluto
  fi
  julia --threads=auto --project=$PLUTO_PROJECT -e "
    $REVISE
    import Pluto;
    notebook_to_open = \"$1\"
    if length(notebook_to_open) == 0
      Pluto.run($PLUTO_ARGS)
    else
      @info \"opening notebook\" notebook_to_open
      Pluto.run(notebook=notebook_to_open, $PLUTO_ARGS)
    end
  "
}
```

It checks if the folder `$HOME/Projects/Pluto.jl` exists and load Pluto from it if so otherwise use the global `@pluto` project. If a path is given, load it with the `notebook` kwarg.

**Comment by mgkuhn** [comment]
The [`-e include(popfirst!(ARGS))` trick](https://docs.julialang.org/en/v1/manual/faq/#man-scripting) is needed to allow Pluto to catch Ctrl-C and terminate gracefully.

(Does anyone know of a good reason why `julia script.jl` doesn't throw [InterruptException](https://docs.julialang.org/en/v1/base/base/#Core.InterruptException) when you try to terminate it with Ctrl-C/SIGINT?)
